experimental/torch_xla2/torch_xla2/ops/jaten.py

```diff
@@ -1159,126 +1159,117 @@ def _aten_relu(self):
-
-  Additional high padding could be required when ceil mode is set.
-  """
-  ceil_mode_padding = []
-  for i in range(len(padding)):
-    left_padding = padding[i]
-    right_padding = left_padding
-
-    input_size = input_shape[2 + i]
-    output_size_rem = (input_size + 2 * left_padding - kernel_size[i]) % stride[
-        i
-    ]
-    if ceil_mode and output_size_rem != 0:
-      extra_padding = stride[i] - output_size_rem
-      new_output_size = (
-          input_size
-          + left_padding
-          + right_padding
-          + extra_padding
-          - kernel_size[i]
-          + stride[i]
-          - 1
-      ) // stride[i] + 1
-      # Ensure that the last pooling starts inside the image.
-      size_to_compare = input_size + left_padding
-
-      if (new_output_size - 1) * stride[i] < size_to_compare:
-        right_padding += extra_padding
-
-    ceil_mode_padding.append((left_padding, right_padding))
-  return ceil_mode_padding
-
+    ceil_mode_padding = []
+    for i in range(len(padding)):
+        left_padding = padding[i]
+        right_padding = left_padding
+
+        input_size = input_shape[2 + i]
+        effective_kernel_size = (kernel_size[i] - 1) * dilation[i] + 1
+        output_size_rem = (input_size + 2 * left_padding - effective_kernel_size) % stride[i]
+        if ceil_mode and output_size_rem != 0:
+            extra_padding = stride[i] - output_size_rem
+            right_padding += extra_padding
+
+        ceil_mode_padding.append((left_padding, right_padding))
+    return ceil_mode_padding
+
+def create_indices(x):
+    batch_size, num_channels, height, width = x.shape
+    size_per_channel = height * width
+    indices_per_channel = jnp.arange(size_per_channel).reshape(height, width)
+    indices = indices_per_channel[None, None, :, :]
+    indices = jnp.tile(indices, (batch_size, num_channels, 1, 1))
+    
+    return indices
 
+#Todo - fix for 1153+1 test - returned tensor shape doesn't match the expected shape - the value matches
 @op(torch.ops.aten.max_pool2d_with_indices)
 @op(torch.ops.aten.max_pool3d_with_indices)
 def _aten_max_pool2d_with_indices(
-  inputs, kernel_size, strides, padding=0, dilation=1, ceil_mode=False
+  inputs, kernel_size, strides=None, padding=0, dilation=1, ceil_mode=False
 ):
-  num_batch_dims = len(inputs.shape) - len(kernel_size) - 1
   kernel_size = tuple(kernel_size)
+  
+  # Set default strides to kernel_size if not provided
+  if not strides:
+      strides = kernel_size
   strides = tuple(strides)
+  
+  # Ensure padding is a list matching kernel dimensions
   if isinstance(padding, int):
-    padding = [padding for _ in range(len(kernel_size))]
+      padding = [padding for _ in range(len(kernel_size))]
+  else:
+      padding = list(padding)
+  
+  # If dilation is an int, make it a tuple
+  if isinstance(dilation, int):
+      dilation = (dilation,) * len(kernel_size)
+  else:
+      dilation = tuple(dilation)
+  
+  if inputs.ndim == 4:
```

Comment:
Looks like you want to reshape to rank = 4?
`inputs = jnp.expand_dims(input, jnp.arange( 4 - inputs.ndim))`

Reply:
that's correct - well pointed, going to incorporate it